docs/source/spec/xml.rst

```diff
@@ -1,26 +1,598 @@
+        <values>
+            <Item>example1</Item>
+            <Item>example2</Item>
+            <Item>example3</Item>
+        </values>
+    </Foo>
+
+
+Flattened list serialization
+----------------------------
+
+The :ref:`xmlflattened-trait` can be used to unwrap the values of list/set
+into a containing structure/union. The name of the elements repeated within
+the structure/union is based on the structure/union member name. For
+example, given the following:
+
+.. code-block:: smithy
+
+    structure Foo {
+        @xmlFlattened
+        flat: MyList,
+    }
+
+The XML serialization of ``Foo`` is:
+
+.. code-block:: xml
+
+    <Foo>
+        <flat>example1</flat>
+        <flat>example2</flat>
+        <flat>example3</flat>
+    </Foo>
+
+The ``xmlName`` trait applied to the structure/union member is used to change
+the name of the repeated XML element. For example, given the following:
+
+.. code-block:: smithy
+
+    union Choice {
+        @xmlFlattened
+        @xmlName("Hi")
+        flat: MySet,
+    }
+
+    set MySet {
+        member: String
+    }
+
+The XML serialization of ``Choice`` is:
+
+.. code-block:: xml
+
+    <Choice>
+        <Hi>example1</Hi>
+        <Hi>example2</Hi>
+        <Hi>example3</Hi>
+    </Choice>
+
+The ``xmlName`` trait applied to the member of a list has no effect when
+serializing a flattened list into a structure/union. For example, given the
+following:
+
+.. code-block:: smithy
+
+    union Choice {
+        @xmlFlattened
+        flat: MySet,
+    }
+
+    set MySet {
+        @xmlName("Hi")
+        member: String
+    }
+
+The XML serialization of ``Choice`` is:
+
+.. code-block:: xml
+
+    <Choice>
+        <flat>example1</flat>
+        <flat>example2</flat>
+        <flat>example3</flat>
+    </Choice>
+
+
+Map serialization
+=================
+
+Map shapes can be serialized as wrapped maps (the default behavior) or
+flattened maps.
+
+
+Wrapped map serialization
+-------------------------
+
+A wrapped map is serialized in an XML element where each value is
+serialized in a nested element named ``entry`` that contains a nested
+``key`` and ``value`` element. For example, given the following:
+
+.. code-block:: smithy
+
+    structure Foo {
+        values: MyMap
+    }
+
+    list MyMap {
+        key: String,
+        value: String,
+    }
+
+The XML serialization of ``Foo`` is:
+
+.. code-block:: xml
+
+    <Foo>
+        <values>
+            <entry>
+                <key>example-key1</key>
+                <value>example1</value>
+            </entry>
+            <entry>
+                <key>example-key2</key>
+                <value>example2</value>
+            </entry>
+        </values>
+    </Foo>
+
+The :ref:`xmlname-trait` can be applied to the key and value members of a map
+to change the nested element names.  For example, given the following:
+
+.. code-block:: smithy
+
+    structure Foo {
+        values: MyMap
+    }
+
+    list MyMap {
+        @xmlName("Name")
```

Comment:
Mismatched with example below using `"Attribute"`.

Reply:
Fixed in the prior rebase